components/kyma-environment-broker/internal/orchestration/manager/manager.go

```diff
@@ -218,6 +237,23 @@ func (m *orchestrationManager) waitForCompletion(o *internal.Orchestration, stra
 		if found {
 			numberOfNotFinished += numberOfPending
 		}
+		numberOfRetrying, found := stats[orchestration.Retrying]
+		if found {
+			// handle the retrying ops during in progress orchestration
+			// use the existing resolved policy in op
+			numberOfNotFinished += numberOfRetrying
+			ops, err := m.factory.RetryOperations(o.OrchestrationID, o.Parameters.Strategy.Schedule, orchestration.MaintenancePolicy{}, false)
```

Comment:
In `waitForCompletion`, this is to handle the retry triggered during the in progress orchestration, so I marked `updateMWindow bool` as `false` for the `m.factory.RetryOperations` here, then the policy will not be used inside, so an empty policy is set here.. (maintenance window time will not be refreshed again during the in progress retry, it will use the resolved window time when the orchestration starts).

Reply:
I mislooked, this absolutely makes sense! 👍